File: src/api/project.ts
// project 命令薄包装。
// 对齐 src-tauri/src/commands/project.rs：
// list_projects / get_project / create_project / update_project / delete_project。

import type { Project, ProjectInput } from '@/types/project'
import { invokeCmd } from './invoke'

/** 列出全部项目，按 id 升序 */
export function listProjects(): Promise<Project[]> {
  return invokeCmd<Project[]>('list_projects')
}

/** 按 id 取项目 */
export function getProject(id: number): Promise<Project> {
  return invokeCmd<Project>('get_project', { id })
}

/** 新建项目。input 字段须用 snake_case（对齐 Rust ProjectInput serde） */
export function createProject(input: ProjectInput): Promise<Project> {
  return invokeCmd<Project>('create_project', { input })
}

/** 更新项目（全量覆盖字段） */
export function updateProject(id: number, input: ProjectInput): Promise<Project> {
  return invokeCmd<Project>('update_project', { id, input })
}

/** 删除项目 */
export function deleteProject(id: number): Promise<void> {
  return invokeCmd<void>('delete_project', { id })
}

/**
 * 按给定顺序批量重排项目 sort_order（拖拽排序后调用）。
 * @param ids 项目 id 数组，下标即新 sort_order
 */
export function reorderProjects(ids: number[]): Promise<void> {
  return invokeCmd<void>('reorder_projects', { ids })
}

File: src-tauri/src/commands/project.rs
//! project 命令薄层
//!
//! 签名对齐 docs/03-命令清单.md。从 AppHandle 取 pool，转调 ProjectService。

use tauri::{AppHandle, Runtime};

use crate::db;
use crate::error::AppResult;
use crate::models::{Project, ProjectInput};
use crate::services::ProjectService;

#[tauri::command]
pub async fn list_projects<R: Runtime>(app: AppHandle<R>) -> AppResult<Vec<Project>> {
    let pool = db::pool(&app)?;
    ProjectService::list(&pool).await
}

#[tauri::command]
pub async fn get_project<R: Runtime>(app: AppHandle<R>, id: i64) -> AppResult<Project> {
    let pool = db::pool(&app)?;
    ProjectService::get(&pool, id).await
}

#[tauri::command]
pub async fn create_project<R: Runtime>(
    app: AppHandle<R>,
    input: ProjectInput,
) -> AppResult<Project> {
    let pool = db::pool(&app)?;
    ProjectService::create(&pool, input).await
}

#[tauri::command]
pub async fn update_project<R: Runtime>(
    app: AppHandle<R>,
    id: i64,
    input: ProjectInput,
) -> AppResult<Project> {
    let pool = db::pool(&app)?;
    ProjectService::update(&pool, id, input).await
}

#[tauri::command]
pub async fn delete_project<R: Runtime>(app: AppHandle<R>, id: i64) -> AppResult<()> {
    let pool = db::pool(&app)?;
    ProjectService::delete(&pool, id).await
}

/// 按 ids 顺序批量重排项目 sort_order（前端拖拽排序后调用）。
#[tauri::command]
pub async fn reorder_projects<R: Runtime>(
    app: AppHandle<R>,
    ids: Vec<i64>,
) -> AppResult<()> {
    let pool = db::pool(&app)?;
    ProjectService::reorder(&pool, ids).await
}
